src/Rules/PhpDoc/VarTagTypeRuleHelper.php

```diff
@@ -151,42 +160,87 @@ private function shouldVarTagTypeBeReported(Node\Expr $expr, Type $type, Type $v
 			}
 		}
 
-		return $this->checkType($type, $varTagType);
+		return $this->checkType($scope, $type, $varTagType);
 	}
 
-	private function checkType(Type $type, Type $varTagType, int $depth = 0): bool
+	private function checkType(Scope $scope, Type $type, Type $varTagType, int $depth = 0): bool
 	{
 		if ($this->strictWideningCheck) {
-			return !$type->isSuperTypeOf($varTagType)->yes();
+			return !$this->isSuperTypeOfVarType($scope, $type, $varTagType);
 		}
 
 		if ($type->isConstantArray()->yes()) {
 			if ($type->isIterableAtLeastOnce()->no()) {
 				$type = new ArrayType(new MixedType(), new MixedType());
-				return $type->isSuperTypeOf($varTagType)->no();
+				return !$this->isAtLeastMaybeSuperTypeOfVarType($scope, $type, $varTagType);
 			}
 		}
 
 		if ($type->isIterable()->yes() && $varTagType->isIterable()->yes()) {
-			if ($type->isSuperTypeOf($varTagType)->no()) {
+			if (!$this->isAtLeastMaybeSuperTypeOfVarType($scope, $type, $varTagType)) {
 				return true;
 			}
 
 			$innerType = $type->getIterableValueType();
 			$innerVarTagType = $varTagType->getIterableValueType();
 
 			if ($type->equals($innerType) || $varTagType->equals($innerVarTagType)) {
-				return !$innerType->isSuperTypeOf($innerVarTagType)->yes();
+				return !$this->isSuperTypeOfVarType($scope, $innerType, $innerVarTagType);
 			}
 
-			return $this->checkType($innerType, $innerVarTagType, $depth + 1);
+			return $this->checkType($scope, $innerType, $innerVarTagType, $depth + 1);
 		}
 
-		if ($type->isConstantValue()->yes() && $depth === 0) {
-			return $type->isSuperTypeOf($varTagType)->no();
+		if ($depth === 0 && $type->isConstantValue()->yes()) {
+			return !$this->isAtLeastMaybeSuperTypeOfVarType($scope, $type, $varTagType);
 		}
 
-		return !$type->isSuperTypeOf($varTagType)->yes();
+		return !$this->isSuperTypeOfVarType($scope, $type, $varTagType);
+	}
+
+	private function isSuperTypeOfVarType(Scope $scope, Type $type, Type $varTagType): bool
+	{
+		if ($type->isSuperTypeOf($varTagType)->yes()) {
+			return true;
+		}
+
+		try {
+			$type = $this->typeNodeResolver->resolve($type->toPhpDocNode(), $this->createNameScope($scope));
+		} catch (NameScopeAlreadyBeingCreatedException) {
+			return false;
```

Comment:
Does this mean it's going to be reported, or it's not going to be reported? I'd rather opt for failing silently (to not report an error).

Reply:
I didn't know exactly what this exception means.

Before my PR we were doing just `$type->isSuperTypeOf($varTagType)->no()`.

> Does this mean it's going to be reported,

Yes. But the error was also reported before this PR. This won't introduce new error.
That's why I didn't chose the failing silently solution.

Now, we doing
- If `$type->isSuperTypeOf($varTagType)->no()` was OK before, early return
- Try to create a namescope
	- If we cannot create a namescope => Report an error (But this error is currently already reported by PHPStan because the check `$type->isSuperTypeOf($varTagType)->no()` was not ok).
	- If we can create a namescope => use it to remove some false-positive.

Reply:
Do you still prefer, I return `true` ?

Reply:
Done